docs/services/shippingservice.md

```diff
@@ -0,0 +1,146 @@
+# Shipping Service
+
+This service is responsible for providing shipping information including pricing
+and tracking information, when requested from Checkout Service.
+
+Shipping service is built primarily with Tonic, Reqwest, and opentelemtry. Other
+sub-dependencies are included in `Cargo.toml`.
+
+Depending on your framework and runtime, you may consider consulting
+[rust docs](https://opentelemetry.io/docs/instrumentation/rust/) to supplement.
+You'll find examples of async and sync spans in quote requests and tracking ID's
+respectively.
+
+The `build.rs` supports development outside docker, given a rust installation.
+Otherwise, consider building with `docker compose` to edit / assess changes as needed.
+
+[Shipping service source](../../src/shippingservice/)
+
+## Traces
+
+### Initialize tracer provider
+
+The OpenTelemetry SDK is initialized from `main`.
+
+```rust
+fn init_tracer() -> Result<sdktrace::Tracer, TraceError> {
+    global::set_text_map_propagator(TraceContextPropagator::new());
+    opentelemetry_otlp::new_pipeline()
+        .tracing()
+        .with_exporter(
+            opentelemetry_otlp::new_exporter()
+                .tonic()
+                .with_endpoint(format!(
+                    "{}{}",
+                    env::var("OTEL_EXPORTER_OTLP_TRACES_ENDPOINT")
+                        .unwrap_or_else(|_| "http://otelcol:4317".to_string()),
+                    "/v1/traces"
+                )), // TODO: assume this ^ is true from config when opentelemetry crate > v0.17.0
+                    // https://github.com/open-telemetry/opentelemetry-rust/pull/806 includes the environment variable.
+        )
+        .install_batch(opentelemetry::runtime::Tokio)
+}
+```
+
+Spans and other metrics are created in this example throughout `tokio` async runtimes
+found within [`tonic` server functions](https://github.com/hyperium/tonic/blob/master/examples/helloworld-tutorial.md#writing-our-server).
+Be mindful of async runtime, [context guards](https://docs.rs/opentelemetry/latest/opentelemetry/struct.ContextGuard.html),
+and inability to [move and clone `spans`](https://stackoverflow.com/questions/25649423/sending-trait-objects-between-threads-in-rust)
+when replicating from these samples.
+
+### Adding GRPC/HTTP instrumentation
+
+This service receives HTTP requests, which are instrumented in the middleware.
```

Comment:
```suggestion
This service receives gRPC requests
```

Is thie following true for the `shippingservice`?
```
 which are instrumented in the middleware.
```
I know that this is valid for PHP, but not sure if that applies to Rust

Reply:
I would consider [tonic](https://github.com/hyperium/tonic) in this case to be acting as middleware.

Reply:
But it is gRPC right?

Reply:
It receives gRPC and sends HTTP